http-update.md

```diff
@@ -0,0 +1,24 @@
+# Triggering version updates over HTTP
+
+To queue an update of your package you can use the `POST /api/packages/:packageName/update` endpoint.
+
+## `POST /api/packages/:packageName/update`
+
+Queues an update for the specified package.
+
+Query params:
+`secret`: string (optional) provide the secret as query
+`header`: string (optional) provide which header is used to check the secret (must start with X-)
+
+Body params: (application/x-www-form-urlencoded, multipart/form-data or application/json)
+`secret`: string (optional) provide the secret as body param
+
+## `POST /api/packages/:packageName/update/github`
+
+Queues an update for the specified package. Compatible with GitHub webhooks and only triggers on `create` events. Must pass secret as query param and not in GitHub webhook settings.
```

Comment:
I wanted to implement github properly at first, but vibe.d with rest interface doesn't give me a way to get the full request body as string, which I need for the SHA1 HMAC verification with the GitHub secret field (at least I didn't find any way)

Reply:
Also I don't think any of the other dub API currently has rate limiting yet. This will only queue a recheck and is basically the same as spamming the "check for updates" button on your project page

Reply:
> I wanted to implement github properly at first,

Don't reinvent the wheel -> https://github.com/dlang/dlang-bot

Reply:
> basically the same as spamming the "check for updates" button on your project page

Fair point, but by providing hooks we make it a lot easier for people to spam / run us into the rate-limits of GitHub.

Reply:
> > I wanted to implement github properly at first,
> 
> Don't reinvent the wheel -> [dlang/dlang-bot](https://github.com/dlang/dlang-bot)

I more accurately meant I wanted to support the "Secret" field on GitHub which sends a Signature of the body text as SHA1 HMAC with the Secret as password, but with vibe.d the rest interface implementation already ate the body so I couldn't access the raw string from code to hash. For hashing the stdlib provides everything needed

Reply:
Regarding the rate limiting, if we have web hook support, I would propose to stretch the polling over a much larger period than now (e.g. 6 hours vs. 30 minutes), so that it merely acts as a fallback in case the web hooks fail. This should reduce the risk of running into the rate limit considerable when compared to the current state.

Non-webhook repositories could be kept at the 30 min. polling interval, of course.

Regarding the signature issue, I believe this should work using an `@before` parameter that reads the body, validates the signature, and then parses and returns the JSON.